rest/route/github.go

```diff
@@ -337,14 +337,29 @@ func (gh *githubHookApi) handleGitTag(ctx context.Context, event *github.PushEve
 		}))
 		return err
 	}
-
+	token, err := gh.settings.GetGithubOauthToken()
+	if err != nil {
+		return errors.New("error getting github token")
+	}
 	pusher := event.GetPusher().GetName()
 	tag := model.GitTag{
 		Tag:    strings.TrimPrefix(event.GetRef(), refTags),
 		Pusher: pusher,
 	}
 	ownerAndRepo := strings.Split(event.Repo.GetFullName(), "/")
-	hash := event.HeadCommit.GetID()
+	hash, err := thirdparty.GetTaggedCommitFromGithub(ctx, token, ownerAndRepo[0], ownerAndRepo[1], tag.Tag)
```

Comment:
Would it make sense to not do this check if event.HeadCommit.GetID() exists?

Reply:
There have been a couple of instances of the ID at HeadCommit not being what we'd expect. Github doesn't even document HeadCommit so there's really not a guarantee on what we should expect.